metaflow/plugins/pypi/micromamba.py

```diff
@@ -26,10 +27,15 @@ def __init__(self):
         # micromamba is a tiny version of the mamba package manager and comes with
         # metaflow specific performance enhancements.
 
-        # TODO: Fix micromamba deteection before shipping
         self.bin = which(
-            os.environ.get("METAFLOW_PATH_TO_MICROMAMBA") or "./micromamba"
-        )
+            os.environ.get("METAFLOW_PATH_TO_MICROMAMBA") or "micromamba"
+        ) or which("./bin/micromamba")
+        if self.bin is None:
+            # Try installing micromamba
+            print("Could not find micromamba installation.")
+            _install_micromamba()
```

Comment:
one big question here is do we need to install micromamba, or can we simply provide users an error and some direction on how to install it?

Thinking that downloading binaries as part of a setup process might not jive well with some organisation policies, even though it is gated by the `which` checks.